src/nvim/aucmd.c

```diff
@@ -50,12 +52,46 @@ void aucmd_schedule_focusgained(bool gained)
 static void do_autocmd_focusgained(bool gained)
 {
   static bool recursive = false;
+  static Timestamp last_time = (time_t)0;
+  bool need_redraw = false;
 
   if (recursive) {
     return;  // disallow recursion
   }
   recursive = true;
-  apply_autocmds((gained ? EVENT_FOCUSGAINED : EVENT_FOCUSLOST),
-                 NULL, NULL, false, curbuf);
+  need_redraw |= apply_autocmds((gained ? EVENT_FOCUSGAINED : EVENT_FOCUSLOST),
+                                NULL, NULL, false, curbuf);
+
+  // When activated: Check if any file was modified outside of Vim.
+  // Only do this when not done within the last two seconds (could get
+  // several events in a row).
+
```

Comment:
remove the empty line, also mention that it makes sense to wait at least one second since on some filesystems the timestamp granularity is in second.

Reply:
FAT32 has a 2 second granularity for modification time.  Most contemporary file systems are much more granular.